### PR TITLE
Make sure there is a maintainer for every testcase

### PR DESCRIPTION
**Pull request by emanchado**
Now that we can specify maintainers at the test case level, we want to
allow specifying _only_ at that level and not at the suite level. That
failed before, so fix that and make sure that every test case has a
maintainer, either from the case itself, or from its containing suite.

### CODE REVIEW

File: kpet/data.py [hunk 1]
```diff
@@ -370,6 +370,8 @@ def __init__(self, data):
                     'The suite has neither name nor description specified.')
             self.name = self.description
         self.validate_case_ids()
+        if self.maintainers is None:
+            self.maintainers = []
```
Comment (spbnick) [inline kpet/data.py:387]: Can you make the above two changes a separate commit?

File: kpet/data.py [hunk 2]
```diff
@@ -563,6 +565,14 @@ def get_sets_from_regex_list(regex_list):
...
                                   format(suite.name, case.name))
 
+                # Make sure there's at least a maintainer (suite- or
+                # case-level)
+                if not case.maintainers and not suite.maintainers:
+                    raise Invalid("There has to be a maintainer for each "
+                                  "test, either at suite level or at case "
+                                  "level; in suite: {}\ncase: {}".
+                                  format(suite.name, case.name))
+
```
Comment (spbnick) [inline kpet/data.py]: This is an inappropriate place for such a check, if only because it has nothing to do with sets. Instead please put this into `kpet.data.Suite.__init__()`

**Review by spbnick — CHANGES_REQUESTED** [review]
Apart from the inline comments, could you please either rework or even completely drop the test? It's too complex for such a small piece of functionality. I think the test could be done entirely in code, with only minimal template handling only the maintainers.

File: kpet/data.py [hunk 3]
```diff
@@ -457,6 +457,20 @@ def validate_host_type_regex(self):
...
+            schema.Invalid when a test case doesn't have any maintainers
+        """
+        for suite in self.suites or []:
+            for case in suite.cases or []:
+                if not case.maintainers and not suite.maintainers:
+                    raise Invalid("There has to be a maintainer for each "
+                                  "test, either at suite level or at case "
+                                  "level; in suite: {}\ncase: {}".
+                                  format(suite.name, case.name))
+
```
Comment (spbnick) [inline kpet/data.py]: Could you please put this into `kpet.data.Suite.__init__()` instead? It only concerns the test suites separately, we don't need the context of the whole database to check cases have maintainers, only the suite context.

**Review by spbnick — CHANGES_REQUESTED** [review]

**Comment by spbnick** [comment]
Looks good! Thank you, Esteban :)
@cki-bot, please test!

**Comment by emanchado** [comment]
Again, the failures seem very random and unrelated to my change 😩
